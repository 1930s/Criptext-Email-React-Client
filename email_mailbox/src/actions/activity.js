import { Activity } from './types';

export const startLoadSync = () => {
  return {
    type: Activity.START_LOAD_SYNC
  };
};

export const stopLoadSync = () => {
  return {
    type: Activity.STOP_LOAD_SYNC
  };
};

export const startLoadThread = () => {
  return {
    type: Activity.START_LOAD_THREAD
  };
};

export const stopLoadThread = () => {
  return {
    type: Activity.STOP_LOAD_THREAD
  };
};
