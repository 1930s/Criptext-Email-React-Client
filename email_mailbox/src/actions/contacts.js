import { Contact } from './types';
import * as db from '../utils/electronInterface';

export const addContacts = contacts => {
  return {
    type: Contact.ADD_BATCH,
    contacts: contacts
  };
};

export const loadContacts = ids => {
  return async dispatch => {
    try {
      const response = await db.getContactByIds(ids);
      const contacts = {};
      response.forEach(element => {
        contacts[element.id] = element;
      });
      dispatch(addContacts(contacts));
    } catch (e) {
      // TO DO
    }
  };
};
