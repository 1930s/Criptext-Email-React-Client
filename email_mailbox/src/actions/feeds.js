import { Feed, Thread } from './types';

export const addFeeds = feeds => {
  return {
    type: Feed.ADD_BATCH,
    feeds: feeds
  };
};

export const muteNotifications = threadId => {
  return {
    type: Thread.MUTE_THREAD,
    targetThread: threadId
  };
};

export const toggleMuteFeed = feedId => {
  return {
    type: Feed.TOGGLE_MUTE,
    targetFeed: feedId
  };
};

export const removeFeed = feedId => {
  return {
    type: Feed.REMOVE,
    targetFeed: feedId
  };
};

export const selectFeed = feedId => {
  return {
    type: Feed.SELECT,
    selectedFeed: feedId
  };
};

export const loadFeeds = () => {
  return async dispatch => {
    try {
      const response = await fetch('/feeds.json');
      const json = await response.json();
      const feeds = json.feeds;
      dispatch(addFeeds(feeds));
    } catch (e) {
      // TO DO
    }
  };
};
