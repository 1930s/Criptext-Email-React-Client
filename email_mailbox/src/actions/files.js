import { File } from './types';
import * as db from '../utils/electronInterface';

export const addFiles = files => {
  return {
    type: File.ADD_BATCH,
    files
  };
};

export const loadFiles = tokens => {
  return async dispatch => {
    try {
      const response = await db.getFilesByTokens(tokens);
      const files = response.reduce(
        (result, file) => ({
          ...result,
          [file.token]: file
        }),
        {}
      );
      dispatch(addFiles(files));
    } catch (e) {
      // TO DO
    }
  };
};
