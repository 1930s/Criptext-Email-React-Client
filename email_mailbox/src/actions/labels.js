import { Label } from './types';
import { getAllLabels } from '../utils/electronInterface';
const inLabel = 10;

export const addLabels = labels => {
  return {
    type: Label.ADD_BATCH,
    labels: labels
  };
};

export const addLabel = label => {
  const labels = {
    inLabel: {
      id: inLabel,
      text: label
    }
  };
  return {
    type: Label.ADD_BATCH,
    labels: labels
  };
};

export const loadLabels = () => {
  return async dispatch => {
    try {
      const response = await getAllLabels();
      const labels = {};
      response.forEach(element => {
        labels[element.id] = {
          id: element.id,
          color: element.color,
          text: element.text
        };
      });
      dispatch(addLabels(labels));
    } catch (e) {
      // TO DO
    }
  };
};
