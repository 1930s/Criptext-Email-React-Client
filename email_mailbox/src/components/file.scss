@import "./../styles/index.scss";

.file-container{
  align-items: center;
  border: 1px solid #e7e5e5;
  border-radius: 6px;
  display: flex;
  height: 38px;
  margin-right: 20px;
  position: relative;
  width: 256px;

  &:hover{
    background-color: #f7f6f6;

    .file-detail-size{
      color: #f7f6f6;
    }
    button{
      display: flex;
    }
  }

  button{
    background: transparent;
    display: none;
    position: absolute;
    right: 20px;
    
    i{
      color: #89909a;
      font-size: 13px;
    }
  }
}

.file-content-icon{
  margin-left: 12px;
  position: relative;

  >div{
    align-items: center;
    border-radius: 4px;
    display: flex;
    height: 22px;
    width: 22px;

    i{
      color: white;
      margin: 0 auto;
    }
  }

  .icon-container-file-default{
    background-color: #4a90e2;
  }
  
}

.file-content-detail{
  display: flex;
  flex-grow: 1;
  margin-left: 10px;
  padding-right: 14px;

  .file-detail-name{
    display: flex;
    flex-grow: 1;
    font-size: 14px;
    font-weight: 700;
  }

  .file-detail-size{
    color: #9b9b9b;
    flex-grow: 0;
    font-size: 13px;
  }
}