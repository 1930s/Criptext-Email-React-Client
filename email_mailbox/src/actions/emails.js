import { Email } from './types';
import {
  getEmailsByThreadId,
  updateUnreadEmailByThreadId,
  setMuteEmailById,
  setUnreadEmailById,
  updateEmail,
  unsendEmailEvent
} from '../utils/electronInterface';
import { loadContacts } from './contacts';
import { updateLabelSuccess } from './labels';
import { EmailStatus, unsentText } from '../utils/const';

export const addEmails = emails => {
  return {
    type: Email.ADD_BATCH,
    emails: emails
  };
};

export const muteNotifications = emailId => {
  return {
    type: Email.MUTE,
    emailId
  };
};

export const markEmailUnreadSuccess = (emailId, unreadValue) => {
  return {
    type: Email.MARK_UNREAD,
    emailId,
    unread: unreadValue
  };
};

export const loadEmails = threadId => {
  return async dispatch => {
    try {
      const response = await getEmailsByThreadId(threadId);
      const data = response.reduce(
        (result, element) => {
          element.from = element.from
            ? element.from.split(',').map(Number)
            : [];
          element.to = element.to ? element.to.split(',').map(Number) : [];
          element.cc = element.cc ? element.cc.split(',').map(Number) : [];
          element.bcc = element.bcc ? element.bcc.split(',').map(Number) : [];
          element.fileTokens = element.fileTokens
            ? element.fileTokens.split(',')
            : [];
          const contactIds = [
            ...element.from,
            ...element.to,
            ...element.cc,
            ...element.bcc
          ];
          return {
            emails: { ...result.emails, [element.id]: element },
            contactIds: new Set([...result.contactIds, ...contactIds])
          };
        },
        { emails: {}, contactIds: new Set() }
      );
      data.contactIds = Array.from(data.contactIds);
      dispatch(loadContacts(data.contactIds));
      dispatch(addEmails(data.emails));
    } catch (e) {
      // TO DO
    }
  };
};

export const muteEmail = (emailId, valueToSet) => {
  return async dispatch => {
    try {
      await setMuteEmailById(emailId, valueToSet);
      dispatch(muteNotifications(emailId));
    } catch (e) {
      // To do
    }
  };
};

export const markEmailUnread = (emailId, valueToSet) => {
  return async dispatch => {
    try {
      await setUnreadEmailById(emailId, valueToSet ? true : false);
      dispatch(markEmailUnreadSuccess(emailId, valueToSet));
    } catch (e) {
      // To do
    }
  };
};

export const updateUnreadEmails = (thread, label) => {
  return async dispatch => {
    try {
      await updateUnreadEmailByThreadId(thread.id, thread.unread);
      if (label) {
        dispatch(updateLabelSuccess(label));
      }
    } catch (e) {
      // To do
    }
  };
};

export const unsendEmail = params => {
  return async dispatch => {
    const { key, emailId } = params;
    try {
      const { status } = await unsendEmailEvent(key);
      if (status === 200) {
        await updateEmail({
          key,
          status: EmailStatus.UNSEND,
          content: unsentText,
          preview: unsentText
        });
        dispatch(unsendEmailOnSuccess(emailId));
      }
    } catch (e) {
      // To do
    }
  };
};

export const unsendEmailOnSuccess = emailId => ({
  type: Email.UNSEND,
  emailId
});
