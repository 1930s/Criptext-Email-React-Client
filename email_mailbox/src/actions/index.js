import { addUsers, addUser } from './users';
import {
  addThreads,
  addThreadLabel,
  addThreadsLabel,
  filterThreadsByUnread,
  loadThreads,
  markThreadsRead,
  moveThreads,
  multiSelectThread,
  muteNotifications,
  selectThread,
  removeThread,
  removeThreadLabel,
  deselectThreads,
  searchThreads,
  selectThreads,
  removeThreads,
  removeThreadsLabel
} from './threads';
import { addEmails, loadEmails } from './emails';
import { addLabels, addLabel, loadLabels } from './labels';
import {
  addFeeds,
  loadFeeds,
  removeFeed,
  selectFeed,
  toggleMuteFeed
} from './feeds';
import { setThreads, loadThreadsSuggestions } from './threadsSuggestions'

export {
  addEmails,
  addFeeds,
  addLabel,
  addLabels,
  addThreads,
  addThreadLabel,
  addThreadsLabel,
  addUser,
  addUsers,
  filterThreadsByUnread,
  loadEmails,
  loadFeeds,
  loadLabels,
  loadThreads,
  loadThreadsSuggestions,
  markThreadsRead,
  moveThreads,
  multiSelectThread,
  muteNotifications,
  removeThread,
  removeThreadLabel,
  deselectThreads,
  searchThreads,
  selectThreads,
  removeFeed,
  removeThreads,
  removeThreadsLabel,
  selectThread,
  selectFeed,
  setThreads,
  toggleMuteFeed
};
