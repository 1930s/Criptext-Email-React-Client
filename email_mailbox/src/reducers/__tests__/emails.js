/* eslint-env node, jest */

import emailReducer from '../emails';
import * as actions from '../../actions/index';
import file from './../../../public/emails.json';
const emails = file.emails;

jest.mock('../../utils/electronInterface');

describe('email actions: ', () => {
  it('should add emails to state', () => {
    const data = {};
    emails.forEach(element => {
      data[element.id] = element;
    });
    const action = actions.addEmails(data);
    const state = emailReducer(undefined, action);
    expect(state).toMatchSnapshot();
  });
});
