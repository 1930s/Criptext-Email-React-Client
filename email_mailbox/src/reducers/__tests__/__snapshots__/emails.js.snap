// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email actions:  Add emails to state 1`] = `
Immutable.Map {
  1: Immutable.Map {
    isMuted: 0,
    threadId: "fSaznwApj6ilX8U",
    preview: "Email content 1",
    subject: "Email subject 1",
    date: 1514231264772,
    unread: 1,
    status: 3,
    secure: 1,
    content: "Email content 1",
    id: 1,
    s3Key: "fSaznwApj6ilX8U",
    key: "fSaznwApj6ilX8U",
  },
  2: Immutable.Map {
    isMuted: 0,
    threadId: "k7ZtKjAXYFSwD",
    preview: "Email content 2",
    subject: "Email subject 2",
    date: 1515624279364,
    unread: 1,
    status: 3,
    secure: 1,
    content: "Email content 2",
    id: 2,
    s3Key: "k7ZtKjAXYFSwD",
    key: "k7ZtKjAXYFSwD",
  },
  3: Immutable.Map {
    isMuted: 0,
    threadId: "HuviK7TKUVK",
    preview: "Email content 3",
    subject: "Email subject 3",
    date: 1512744868344,
    unread: 1,
    status: 3,
    secure: 1,
    content: "Email content 3",
    id: 3,
    s3Key: "HuviK7TKUVK",
    key: "HuviK7TKUVK",
  },
}
`;
